src/non-empty-array.ts

```diff
@@ -0,0 +1,4 @@
+export type NonEmptyArray<T> = [T, ...T[]];
```

Comment:
I read in a book somewhere which was suggested that it is good to avoid names of variables or types with negative connotation

Reply:
Reason being that when we negate a variable with negative connotation, the meaning becomes difficult to decipher e.g. ```!isNotShoes```

Reply:
In this case that's what the actual type is. Is kind of the same [as this one](https://hackage.haskell.org/package/base-4.14.0.0/docs/Data-List-NonEmpty.html) and [this one](https://gcanti.github.io/fp-ts/modules/NonEmptyArray.ts.html).